### PR TITLE
feat(BQStream): add batch support

### PR DESCRIPTION
**Pull request by akashrpo**
## Description of the change

> add batch support

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/bqstream/transform.js [hunk 1]
```diff
@@ -42,4 +50,96 @@ const process = async event => {
...
+        batchEventResponse.destination,
+        true
+      )
+    );
+  });
+
+  return batchedResponseList;
+};
+
+const processRouterDest = async inputs => {
```
Comment (sanpj2292) [inline v0/destinations/bqstream/transform.js:100]: There is already a utility for router transformation(without batching) in `v0/utils/index.js`. Can we see if we can make use of it ?

Comment (akashrpo) [reply v0/destinations/bqstream/transform.js:101]: updated

File: v0/destinations/bqstream/transform.js [hunk 2]
```diff
@@ -42,4 +51,95 @@ const process = async event => {
...
+  const errorRespEvents = checkInvalidRtTfEvents(inputs, DESTINATION);
+  if (errorRespEvents.length > 0) {
+    return errorRespEvents;
+  }
+
+  const eventsChunk = []; // temporary variable to divide payload into chunks
+  const errorRespList = [];
+  await Promise.all(
```
Comment (koladilip) [inline v0/destinations/bqstream/transform.js:109]: This block looks very similar and I have seen many times, is there util for this? if not please create one.

File: v0/destinations/bqstream/transform.js [hunk 3]
```diff
@@ -42,4 +51,95 @@ const process = async event => {
...
+    inputs.map(async event => {
+      try {
+        if (event.message.statusCode) {
+          // already transformed event
+          eventsChunk.push(event);
```
[the remark this answers is not in the document]
Comment (akashrpo) [reply v0/destinations/bqstream/transform.js:114]: async way is not really being used here and can be removed as there aren't any intermediate network calls. pushed the changes

File: v0/destinations/bqstream/transform.js (hunk 2 above)
Comment (akashrpo) [reply v0/destinations/bqstream/transform.js]: we do not have a util function of this yet and removing async way as mentioned below makes this specific to bqstream.

**Review by koladilip — DISMISSED** [review]

**Review by utsabc — DISMISSED** [review]

**Review by koladilip — DISMISSED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by sanpj2292 — APPROVED** [review]

File: src/v0/destinations/bqstream/transform.js [hunk 4]
```diff
@@ -21,7 +21,7 @@ const getInsertIdColValue = (properties, insertIdCol) => {
   return null;
 };
 
-const process = event => {
+const process = async (event) => {
```
Comment (akashrpo) [inline src/v0/destinations/bqstream/transform.js:24]: @ItsSudip async is not really required. This was removed as part of the last review comment.

Comment (ItsSudip) [reply src/v0/destinations/bqstream/transform.js]: Removed the async can you check?

**Review by sanpj2292 — DISMISSED** [review]

**Review by yashasvibajpai — DISMISSED** [review]

**Review by ItsSudip — APPROVED** [review]

File: src/v0/destinations/bqstream/transform.js [hunk 5]
```diff
@@ -41,4 +50,91 @@ const process = async (event) => {
...
+const processRouterDest = (inputs) => {
+  const errorRespEvents = checkInvalidRtTfEvents(inputs, DESTINATION);
+  if (errorRespEvents.length > 0) {
+    return errorRespEvents;
+  }
+
+  const eventsChunk = []; // temporary variable to divide payload into chunks
+  const errorRespList = [];
+  Promise.all(
+    inputs.map(async (event) => {
```
Comment (akashrpo) [inline src/v0/destinations/bqstream/transform.js:109]:
```suggestion
    inputs.map((event) => {
```

Comment (ItsSudip) [reply src/v0/destinations/bqstream/transform.js]: Updated

**Review by ItsSudip — APPROVED** [review]

**Review by sanpj2292 — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Pull request merged**